ui/src/pages/server-types/server-types.tsx

```diff
@@ -0,0 +1,91 @@
+import { AppProfileProps } from '@src/types/api';
+import axios from '@src/utils/axios';
+import React, { useEffect, useState } from 'react';
+import RadioButton from '../../components/radio-button/radio-button';
+
+export const ServerTypes = (): React.ReactElement => {
+  const [selectedServerType, setSelectedServerType] = useState<string>('');
+  const [serverTypes, setServerTypes] = useState<AppProfileProps[]>([]);
+
+  useEffect(() => {
+    axios
+      .get('/spawner-profiles/')
+      .then((response) => {
+        console.log('Full response:', response);
+        return setServerTypes(response.data);
+      })
+      .catch((error) => {
+        console.error('Error fetching data:', error);
+      });
+  }, []);
```

Comment:
You should be using react-query for making API calls, should be no need for a useEffect. See here for an example: https://github.com/nebari-dev/jhub-apps/blob/f3ce17cf3e092476e94e33e2de509d6184ec4593/ui/src/pages/home/apps-grid/apps-grid.tsx#L25